src/TypeDefinition.js

```diff
@@ -404,11 +436,13 @@ export type NavigationDispatch = (
 
 export type NavigationProp<S> = {
   +state: S,
+  addListener: NavigationEventSubscriber,
```

Comment:
let's maybe call this `NavigationNavigatorProp` (and as a follow-up, consider removing the prefix `Navigation` throughout our types)